extensions/fleet/fleet.go

```diff
@@ -0,0 +1,76 @@
+package fleet
+
+import (
+	"time"
+
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	"github.com/rancher/shepherd/clients/rancher"
+	v1 "github.com/rancher/shepherd/clients/rancher/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	FleetGitRepoResourceType = "fleet.cattle.io.gitrepo"
+)
+
+// CreateFleetGitRepo is a "helper" functions that takes a rancher client, and the rke2 cluster config as parameters. This function
+// registers a delete cluster fuction with a wait.WatchWait to ensure the cluster is removed cleanly.
+func CreateFleetGitRepo(client *rancher.Client, gitRepo *v1alpha1.GitRepo) (*v1.SteveAPIObject, error) {
```

Comment:
As CreateFleetGitRepo function involves creating a Fleet GitRepo and includes a retry and cleanup logic, IMO this should be considered an action rather than an extension. 

Could you let me know why we want to have this as an extension? Understanding the reasoning would help clarify whether thats the best approach.

Reply:
messaged offline, lmk what you think but tldr is I think this still fits the definition of extension

Reply:
discussed offline and since the gitrepo and client are the only parameters required and no additional config is required now or in future, we can have this in extensions. thanks! resolving the comment.